_posts/2022-10-24-public-jenkins.markdown

```diff
@@ -0,0 +1,53 @@
+
+This build system act as a continuous integration and continuous deployment (CI / CD) system that triggers the generation of distribution builds along with automated execution of integration, backward compatibility and performance tests for every build. The enhanced process also automatically publishes the artifacts to target distribution channels including but not limited to Docker, S3, cloud front, Maven etc. The plugin developers can on-board their plugin artifacts into OpenSearch distributions by adding the component to the distribution input manifest file. The onboarding plugin will need to provide information such as GitHub repository URL, the branch which needs to be built, which platforms does the plugin support and checks to run on the component such as version check, dependency check, etc. See [example](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/1.3.0/opensearch-1.3.0.yml#L29-L34). The end to end build process uses the information on the file to build, assemble, test and publish the artifacts to corresponding distribution targets.
+
+The build system is deployed using infrastructure as code that uses AWS CDK. The build infrastructure is completely reproducible and consumes [opensearch-ci](https://github.com/opensearch-project/opensearch-ci) repository as its source. The[readme.md](https://github.com/opensearch-project/opensearch-ci/blob/main/README.md) file in [opensearch-ci](https://github.com/opensearch-project/opensearch-ci) repository provides the instructions to reproduce this infrastructure. External contributors can contribute to the build ecosystem by raising the PR on [this](https://github.com/opensearch-project/opensearch-ci) GitHub repository for any future enhancements, bug fixes and new features. 
+
+## Motivation
+
+Before now, setting up the test and build infrastructure was a tedious process that required a lot of manual intervention and development effort to make changes and updates. Also, the infrastructure used for building and releasing the artifacts was not reproducible, and teams were finding it hard to track the status of builds due to limited visibility into the build process. This created discrepancies between the development team in terms of building, testing, and debugging the artifact builds during every release. 
+
+The manual processes also added additional overhead for the development teams to create, own and maintain their own automation build process, mimicking our process. The new system automates all the manual steps involved in creating, managing, maintaining and monitoring the DevOps stack for OpenSearch distribution builds to improve reproducibility along with developer and release velocity.
+
+### OpenSearch CI / CD infrastructure design:
+
+![Figure 1: Jenkins Infrastructure Overview]({{ site.baseurl }}assets/media/blog-images/2022-09-19-public-jenkins.markdown/jenkins.png){: .img-fluid }**Figure 1**: Jenkins Infrastructure Overview
```

Comment:
Change this to the new dates I think?
@gaiksaya

Reply:
Will do! Thanks!